api/envoy/config/cluster/v3/cluster.proto

```diff
@@ -205,6 +205,11 @@ message Cluster {
     // have the same restrictions as cluster name, i.e. it may be arbitrary
     // length. This may be a xdstp:// URL.
     string service_name = 2;
+
+    // If set, all discovery requests for EDS will utilize a single GRPC stream towards the same management server.
+    // Defaults to false. When unset, Envoy will allocate a GRPC stream per EDS cluster which can lead to the exhaustion of
+    // the limit for max concurrent streams in HTTP2.
+    bool multiplex_eds = 3;
```

Comment:
I don't think this should be done by adding a new field to the API like this, especially not an EDS-specific field.  The fact that multiple EDS subscriptions are not currently multiplexed onto a single stream is basically a bug in Envoy, and I think we should just fix the bug.

If you explicitly want to find a way to make this configurable via the API, then a better approach would be to fix #13951.  That would allow a `ConfigSource` to specify whether it's an ADS stream, in which case Envoy will multiplex the EDS subscriptions onto a single ADS stream.  And that approach would work for all resource types, not just EDS.

Reply:
@markdroth the config option was introduced to reduce risk of rolling out the change, as previously merged [fix](https://github.com/envoyproxy/envoy/pull/5272) has broken EDS for bunch of users. I agree is not the best usage of API, as eventually everyone should use the multiplexed version of subscription factory for EDS. Perhaps we could instead guard new feature with runtime flag? We have increased confidence for current bug fix, as it is now covered by integration tests and has been tested in prod environment, but there is still risk involved.

I would like to focus first on fixing EDS bug, as it has significant impact on us. I can further help out with https://github.com/envoyproxy/envoy/issues/13951 but as a separate effort.

Reply:
A runtime flag sounds fine.

I would really prefer not to add an EDS-specific option to the API for this, since it's really not an API issue, it's just an Envoy implementation bug.